### ISSUE TITLE
Change cp.dialog to display alerts in CommandPost, not Frontmost Window

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#505]

**Comment by randomeizer** [comment]
What is this improving, out of curiosity?

**Comment by latenitefilms** [comment]
At the moment if the frontmost app is TextEdit for example, when you do a cp.dialog.displayMessage() it will make the alert appear as if it's from TextEdit.

This is great if you're in Final Cut Pro, but seems to cause issues if you're in CommandPost Preferences, as it doesn't look like AppleScript treats CommandPost as a "normal app" - probably because it doesn't have a dock icon.

I've just noticed a few times when I've pressed something in Preferences that a alert won't appear at frontmost for whatever reason, and it's not obvious where the alert is hiding.

I might also try and rework it the basic cp.dialog.displayMessage in Objective-C land to see if it helps with the other Sierra bugs. Will have a look tonight.
